tests/restapi/conftest.py

```diff
@@ -48,14 +48,10 @@ def server_url():
 
 
 @pytest.fixture
-def restrict_sqlalchemy_queuepool(aiida_profile):
+def restrict_sqlalchemy_queuepool(aiida_profile):  # pylint: disable=unused-argument
     """Create special SQLAlchemy engine for use with QueryBuilder - backend-agnostic"""
-    from aiida.manage.manager import get_manager
-
-    backend_manager = get_manager().get_backend_manager()
-    backend_manager.reset_backend_environment()
-    actual_profile = aiida_profile._manager._profile  # pylint: disable=protected-access
-    backend_manager.load_backend_environment(actual_profile, pool_timeout=1, max_overflow=0)
+    # TODO re-instate this fixture?
```

Comment:
This fixture was explicitly created for a unit test of the REST API. Unless the test can be removed, this should probably not be removed.

Reply:
the tests pass without this fixture though. What I want to understand, to think if/how it should be re-instated, is exactly what the `pool_timeout=1, max_overflow=0` are intended to test, as this was not documented at all

Reply:
> the tests pass without this fixture though

Well the point is that it is creating a situation (an engine with reduced pool timeout) that caused problems in real-life setups. The test made sure that even under those constrained conditions, things still worked. So of course if you remove the constraint it might still work, but you are not testing what you were intending to test.

Reply:
I think your understanding of the fixture and test was incorrect 😬, which was my point of it being poorly documented.
The actual goal has nothing really to do with pool timeouts, it is just checking that threads properly close their database connections (see https://github.com/aiidateam/aiida-core/commit/b9d4bbebcbd1719aa0947549c34a689a6853a1bc).

Anyhow, I've now re-worked it in [`1b98a91` (#5330)](https://github.com/aiidateam/aiida-core/pull/5330/commits/1b98a91ca394ffa2d406fef4078802f8cafc2903).
Rather than trying to hack into the backend's connection creation, the profile's `storage_config` section now allows for an `engine_kwargs` key, which is used by `create_sqlalchemy_engine` to allow any additional kwargs to be passed to `sqlalchemy.create_engine`

Reply:
Note, its still not ideal, because the restapi is essentially hard-coded to the `psql_dos` backend, relying on sqlalchemy's `scoped_session` to handle thread safety.
Going forward, I'm going to look into making `get_manager` thread local, which should mean you can properly use AiiDA in a multi-threaded mode

Reply:
> I think your understanding of the fixture and test was incorrect grimacing,

Are you really sure. I remember the PR that added that fixture. The problem was as you say that the REST API threads weren't properly closing the database connections. This would cause them to accumulate and at some point the calls would fail since there were no more connections allowed. To test the fix, @CasperWA wanted to make sure that there was a test that would try to hit the limit. If this were to be done with the standard settings, it would take way too long, so to speed up the process, the engine was configured such that it would hit the error almost instantly in case of a regression.

Reply:
yep thats what it does now 👍

Reply:
![200](https://user-images.githubusercontent.com/43357585/153366553-3d2d4fe1-efc7-4152-9039-cc28ab9ca09d.gif)